sql/core/src/main/scala/org/apache/spark/sql/execution/Sort.scala

```diff
@@ -93,4 +98,63 @@ case class Sort(
       sortedIterator
     }
   }
+
+  override def upstream(): RDD[InternalRow] = {
+    child.asInstanceOf[CodegenSupport].upstream()
+  }
+
+  // Name of sorter variable used in codegen.
+  private var sorterVariable: String = _
```

Comment:
it's ok here as discussed offline. i just found mutable state in here as a way to pass variable names through pretty brittle. maybe good to have a more general abstraction for this in codegen, but not that big of a deal right now/